solution/backend/resources/views/resources.py

```diff
@@ -34,6 +42,97 @@
+            default=None,
+        )
+
+    def get_gov_results(self, query, page):
+        key = 'M1igE4Qcfo8LLQr7o_I9KLA6qkybmlC9IRhVCCbFbl4='
+        offset = (page - 1) * self.limit
+        results = []
+        rstring = f'https://search.usa.gov/api/v2/search/?affiliate=reg-pilot-cms-test&access_key={key}' \
+                  f'&query={query}&limit={self.limit}&offset={offset}'
+        gov_results = json.loads(requests.get(rstring).text)
+        if "errors" in gov_results or (gov_results["web"]["total"] == 0 and not gov_results["web"]["results"]):
+            raise NotFound("Invalid page")
+
+        for gov_result in gov_results['web']['results']:
+            result = {
+                'name': gov_result['title'],
+                'snippet': gov_result['snippet'],
+                'url': gov_result['url']
+            }
+            results.append(result)
+        self.gov_results = {
+            "total": gov_results["web"]["total"],
+            "results": results,
+        }
+
+    def get_queryset(self, urls):
+
+        locations_prefetch = AbstractLocation.objects.all().select_subclasses()
+        category_prefetch = AbstractCategory.objects.all().select_subclasses().select_related("subcategory__parent")
+
+        return AbstractResource.objects \
+            .filter(approved=True) \
+            .annotate(url_annotated=self.get_annotated_url()) \
+            .filter(url_annotated__in=urls) \
+            .select_subclasses().prefetch_related(
+                Prefetch("locations", queryset=locations_prefetch),
+                Prefetch("category", queryset=category_prefetch),
+                Prefetch("related_resources", AbstractResource.objects.all().select_subclasses().prefetch_related(
+                    Prefetch("locations", queryset=locations_prefetch),
+                    Prefetch("category", queryset=category_prefetch),
+                )),
+            )
+
+    def append_snippet(self, queryset, urls):
+        for q in queryset:
+            q.snippet = urls[q.url]
+        return queryset
+
+    def generate_page_url(self, url, page):
+        parse_url = urlparse.urlparse(url)
+        url_parts = parse_url.query.split("&")
+        new_url_parts = list(map(lambda x: re.sub(r"page=\d", f"page={page}", x), url_parts))
+        query = "&".join(new_url_parts)
+        parsed_url = parse_url._replace(query=query)
+        return urlparse.urlunparse(parsed_url)
+
+    def list(self, request, *args, **kwargs):
+        q = self.request.query_params.get("q")
+        page = int(self.request.query_params.get("page", 1))
+        url = request.get_full_path()
+        self.get_gov_results(q, page)
+        urls = dict([(i['url'], i["snippet"]) for i in self.gov_results["results"]])
+        queryset = self.get_queryset(urls.keys())
+        resources = list(queryset)
+        urls = dict([(i['url'], i["snippet"]) for i in self.gov_results["results"]])
```

Comment:
this line got duplicated for assigning the values for URLs at some point.  Just remove it since the proper value for urls is already assigned on 115.